nifi-nar-bundles/nifi-questdb-bundle/nifi-questdb/src/main/java/org/apache/nifi/questdb/embedded/EmbeddedDatabaseManager.java

```diff
@@ -0,0 +1,297 @@
+                    }
+                }
+
+                if (couldMoveOldToBackup) {
+                    try {
+                        ensurePersistLocationIsAccessible();
+                        ensureConnectionEstablished();
+                        ensureTablesAreInPlaceAndHealthy();
+                        successful = true;
+                    } catch (CorruptedDatabaseException ex) {
+                        LOGGER.error("Could not create backup", ex);
+                    }
+                }
+            }
+        } finally {
+            state.set(successful? EmbeddedDatabaseManagerStatus.HEALTHY : EmbeddedDatabaseManagerStatus.CORRUPTED);
+
+            if (!successful) {
+                engine.set(null);
+            }
+
+            databaseStructureLock.writeLock().unlock();
+        }
+
+    }
+
+    private void ensurePersistLocationIsAccessible() throws CorruptedDatabaseException {
+        try {
+            FileUtils.ensureDirectoryExistAndCanAccess(context.getPersistLocationAsPath().toFile());
+        } catch (final Exception e) {
+            final String errorMessage = String.format("Database directory creation failed [%s]", context.getPersistLocationAsPath());
+            LOGGER.error(errorMessage, e.toString());
+            throw new CorruptedDatabaseException(errorMessage, e);
+        }
+    }
+
+    private void ensureConnectionEstablished() throws CorruptedDatabaseException {
+        if (engine.get() != null) {
+            engine.getAndSet(null).close();
+        }
+
+        final String absolutePath = context.getPersistLocationAsFile().getAbsolutePath();
+        final CairoConfiguration configuration = new DefaultCairoConfiguration(absolutePath);
+
+        try {
+            final CairoEngine engine = new CairoEngine(configuration);
+            LOGGER.info("Database connection successful [{}]", absolutePath);
+            this.engine.set(engine);
+        } catch (final Exception e) {
+            LOGGER.error("Database connection failed [{}]", absolutePath, e);
+            throw new CorruptedDatabaseException(e);
+        }
+    }
+
+    private void ensureTablesAreInPlaceAndHealthy() throws CorruptedDatabaseException {
+        final Map<String, File> databaseFiles = Arrays.stream(context.getPersistLocationAsFile().listFiles())
+                .collect(Collectors.toMap(f -> f.getAbsolutePath().substring(context.getPersistLocationAsFile().getAbsolutePath().length() + 1), f -> f));
+        final Client client = getUnmanagedClient();
+
+        try {
+            for (final ManagedTableDefinition tableDefinition : context.getTableDefinitions()) {
+                if (!databaseFiles.containsKey(tableDefinition.getName())) {
+                    try {
+                        LOGGER.info("Creating table {}", tableDefinition.getName());
+                        client.execute(tableDefinition.getDefinition());
+                        LOGGER.info("Table {} is created", tableDefinition.getName());
```

Comment:
This look like they should be debug messages.
```suggestion
                        LOGGER.debug("Creating table {}", tableDefinition.getName());
                        client.execute(tableDefinition.getDefinition());
                        LOGGER.debug("Table {} created", tableDefinition.getName());
```